sway-lib-std/src/ecr.sw

```diff
@@ -1,21 +1,33 @@
 library ecr;
 
-use ::b512::B512;
 use ::address::Address;
+use ::b512::B512;
+use ::constants::ZERO;
+use ::result::Result;
+
+
+pub enum EcRecoverError {
+    UnrecoverableAddress: (),
+    UnrecoverablePublicKey: (),
+}
 
 /// Recover the public key derived from the private key used to sign a message
-pub fn ec_recover(signature: B512, msg_hash: b256) -> B512 {
+pub fn ec_recover(signature: B512, msg_hash: b256) -> Result<B512, EcRecoverError> {
     let public_key = ~B512::new();
 
     asm(buffer: public_key.bytes, sig: signature.bytes, hash: msg_hash) {
         ecr buffer sig hash;
     };
-
-    public_key
+    /// if recovered key is empty
+    if public_key == ~B512::new() {
```

Comment:
This adds the overhead of instantiating a new B512, I think I can improve it by comparing `(public_key)[0] == ZERO && (public_key)[1]`